xbmc/addons/Repository.cpp

```diff
@@ -250,14 +246,23 @@ bool CRepositoryUpdateJob::DoWork()
 
     for (const auto& addon : addons)
     {
-      if (!addon->Props().fanart.empty())
-        textureDB.InvalidateCachedTexture(addon->Props().fanart);
-      if (!addon->Props().icon.empty())
-        textureDB.InvalidateCachedTexture(addon->Props().icon);
+      AddonPtr oldAddon;
+      if (database.GetAddon(addon->ID(), oldAddon) && addon->Version() > oldAddon->Version())
+      {
+        if (!addon->Props().icon.empty())
+        {
+          CLog::Log(LOGDEBUG, "CRepository: invalidating cached art for '%s'", addon->ID().c_str());
```

Comment:
https://github.com/xbmc/xbmc/blob/master/xbmc/addons/Addon.cpp#L187

and we have addons that have no icons. https://github.com/kodi-pvr/pvr.stalker/tree/master/pvr.stalker

so I dont think it's "mandatory"